draft-ietf-quic-http.md

```diff
@@ -1396,22 +1396,25 @@ servers, the initial value of each client setting is the default value.
 For clients using a 1-RTT QUIC connection, the initial value of each server
 setting is the default value. When a 0-RTT QUIC connection is being used, the
 initial value of each server setting is the value used in the previous session.
-Clients MUST store the settings the server provided in the session being resumed
-and MUST comply with stored settings until the current server settings are
-received.  A client can use these initial values to send requests before the
+Clients SHOULD store the settings the server provided in the session being
+resumed and MUST comply with stored settings until the current server settings
+are received.  If the client did not store, or did not receive, server settings
+from the previous session, the initial value of each setting is the default
+value.  A client can use these initial values to send requests before the
 server's SETTINGS frame has arrived.  This removes the need for a client to wait
 for the SETTINGS frame before sending requests.
 
 A server can remember the settings that it advertised, or store an
 integrity-protected copy of the values in the ticket and recover the information
 when accepting 0-RTT data. A server uses the HTTP/3 settings values in
-determining whether to accept 0-RTT data.
+determining whether to accept 0-RTT data.  If the server cannot recover the
+settings from a session being resumed, it MUST NOT accept 0-RTT data.
```

Comment:
by my reading, the paragraph describes to methods: remember or recover. This new clause seems to imply if you can't recover, then remember is not allowed.

Is it more aligned with the intent to say that when a server chooses the recover method, and encodes the data in a ticket, and a client presents a ticket, then the server MUST recover the settings or else reject 0-RTT?

Reply:
I don't think that we're mandating the use of session tickets, even if that is the most common method of implementing these.

I think that "recover" is probably still right, though it might be clearer here if we say "If a server cannot determine that the settings remembered by a client are compatible with its current settings, it MUST reject 0-RTT."

Then we need to define what "compatible" means.